warn/multifile.go

```diff
@@ -0,0 +1,48 @@
+package warn
+
+import (
+	"github.com/bazelbuild/buildtools/build"
+)
+
+// FileReader is a class that can read an arbitrary Starlark file
+// from the repository and cache the results.
+type FileReader struct {
+	cache    map[string]*build.File
+	readFile func(string) ([]byte, error)
+}
+
+// NewFileReader initializes a FileReader instance with an instance of function
+// that can read an arbitrary file in the repository using a path relative
+// to the workspace root.
+func (fr *FileReader) NewFileReader(readFile func(string) ([]byte, error)) {
```

Comment:
this should not be a receiver, but return *FileReader

func  NewFileReader(readFile func(string) ([]byte, error)) *FileReader {

Reply:
Done.